scripts/deployer.ts

```diff
@@ -0,0 +1,280 @@
+import { ethers } from 'hardhat';
+import hre from 'hardhat';
+
+interface DeployedContract {
+  abi: Array<never>;
+  bytecode: string;
+  address: string;
+  block: number;
+  url: string;
+}
+
+interface DeployedData {
+  contracts: {
+    postageStamp: DeployedContract;
+    redistribution: DeployedContract;
+    staking: DeployedContract;
+    priceOracle: DeployedContract;
+    factory: DeployedContract;
+  };
+}
+
+// URL
+const mainnetURL = 'https://gnosisscan.io/address/';
+const testnetURL = 'https://goerli.etherscan.io/address/';
+
+const blockChainVendor = hre.network.name;
+const networkID = hre.network.config.chainId;
+
+async function main(deployedData: DeployedData = testnetData) {
+  let contractData = {
+    addresses: {
+      postageStamp: '',
+      redistribution: '',
+      staking: '',
+      priceOracle: '',
+      factory: '',
+    },
+    blocks: {
+      postageStamp: 0,
+      redistribution: 0,
+      staking: 0,
+      priceOracle: 0,
+      factory: 0,
+    },
+  };
+
+  switch (blockChainVendor) {
+    case 'testnet':
+      contractData = await deployFactory(await JSON.parse(JSON.stringify(testnetData).toString()), contractData);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      deployedData = testnetData;
+      break;
+    case 'mainnet':
+      contractData = await deployFactory(await JSON.parse(JSON.stringify(mainnetData).toString()), contractData);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      deployedData = mainnetData;
+      break;
+    default:
+      contractData = await deployFactory(null, contractData);
+  }
+
+  if (!contractData.addresses.factory) {
+    handleErr('expected factory address');
+  }
+
+  contractData = await deployRedistribution(
+    await deployPriceOracle(await deployPostageStamp(await deployStaking(contractData)))
+  );
+
+  await rolesSetter(contractData);
+
+  await writeResult(deployedData, contractData);
+  await logResult(contractData);
+}
+
+async function deployFactory(deployed: any, contractData: any) {
+  if (deployed == null) {
+    const TestToken = await hre.ethers.getContractFactory('TestToken');
+    const testToken = await TestToken.deploy();
+    await testToken.deployed();
+    contractData.addresses.factory = testToken.address;
+    contractData.blocks.factory = testToken.deployTransaction.blockNumber as number;
+    return contractData;
+  }
+  contractData.addresses.factory = deployed['contracts']['factory']['address'];
+  contractData.blocks.factory = deployed['contracts']['factory']['block'];
+  return contractData;
+}
+
+async function deployStaking(contractData: any) {
+  const StakeRegistryContract = await hre.ethers.getContractFactory('StakeRegistry');
+  const stakeRegistryContract = await StakeRegistryContract.deploy(contractData.addresses.factory, networkID);
+  await stakeRegistryContract.deployed();
+  console.log(stakeRegistryContract.address);
```

Comment:
is it needed to log this? if so it would be good to be more descriptive

Reply:
Agree, this is not needed since it's logged below. The same applies to other functions.